disnake/ext/commands/params.py

```diff
@@ -86,8 +86,13 @@
 else:
     P = TypeVar("P")
 
+
 if sys.version_info >= (3, 10):
     from types import EllipsisType, UnionType
+elif TYPE_CHECKING:
+    EllipsisType = type(Ellipsis)
+    UnionType = Type[Union]
```

Comment:
```suggestion
    UnionType = NoReturn
```
`types.UnionType` is specifically for pipe unions ("union type expressions"), which don't exist in <3.10, so I'd rather have these annotated as `Never`/`NoReturn`.
(weird how pyright doesn't complain about `Union[User, Member]` being used as a key to `dict[UnionType, ...]` even in later versions, but that just makes it easier for us)